webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -209,24 +210,29 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
         },
         { signal: canceler.signal },
       );
+      const total = response.pagination.total ?? 0;
+      const loadedExperiments = response.experiments;
 
-      setExperiments((prevExperiments) => {
+      setExperiments((prev) => {
         if (pagedView) {
-          return response.experiments.map((e) => Loaded(e));
+          return loadedExperiments.map((experiment) => Loaded(experiment));
         }
-        const experimentBeforeCurrentPage = [
-          ...prevExperiments.slice(0, tableOffset),
-          ...Array(Math.max(0, tableOffset - prevExperiments.length)).fill(NotLoaded),
-        ];
 
-        const experimentsAfterCurrentPage = prevExperiments.slice(
-          tableOffset + response.experiments.length,
-        );
-        return [
-          ...experimentBeforeCurrentPage,
-          ...response.experiments.map((e) => Loaded(e)),
-          ...experimentsAfterCurrentPage,
-        ].slice(0, response.pagination.total);
+        let newExperiments = prev;
+
+        // Fill out the loadable experiments array with total count.
+        if (prev.length !== total) {
+          newExperiments = [...prev, ...Array(total - prev.length).fill(NotLoaded)];
```

Comment:
There might be a situation where total is smaller than prev.length and we end up with an invalid array length

Reply:
ah good call, I'll put something in for that case. Thanks!

Reply:
That could be a case but it's not exactly what I had in mind. If I start at page=10, total could be smaller than prev.length

Reply:
https://github.com/determined-ai/determined/assets/40620519/e989cb09-9cc4-40be-9def-dc40a5d267f3

Reply:
ah gotcha. I ended up using `total` as the source of truth upon a successful response so even when filter is applied it should be updated to the correct count (in the case where `prev.length` doesn't match up with `total`)